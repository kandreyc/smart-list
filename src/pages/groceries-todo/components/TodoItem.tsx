import PriceData from "../../../entity/PriceData";
import Product from "../../../entity/Product";

export class ProductPriceData {
    latestPrice: number;
    perCounterpartyPrice: {[id: string]: PriceData};

    constructor () {
        this.latestPrice = 0;
        this.perCounterpartyPrice = {};
    }

    setCounterpartyPrice(counterparty: string, priceData: PriceData) {
        this.perCounterpartyPrice[counterparty] = priceData;
    }
}

class TodoItem {
    id: number;
    generalName: string;
    quantity: number;
    targetProduct?: Product;
    isBought: boolean;
    priceData: ProductPriceData;

    constructor (id: number, generalName: string) {
        this.id = id;
        this.generalName = generalName;
        this.quantity = 1;
        this.isBought = false;
        this.priceData = new ProductPriceData();
    }

    incrementQuantity() {
        this.quantity ++;
    }

    decrementQuantity() {
        if (this.quantity > 1) {
            this.quantity --;
        }
    }
}

export default TodoItem;
