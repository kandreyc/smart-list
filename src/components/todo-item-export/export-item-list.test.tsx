import { shallow } from "enzyme";
import TodoItem from "../todo-item-list/types";
import ExportItemList from "./export-item-list";

type OverridingProps = {
    todoItems: TodoItem[];
};

function createTodoItem(generalName: string) {
    return TodoItem.createTodoItem(Date.now(), generalName);
}

describe("ExportItemList", () => {
    const exportItemList = ({ ...overridingProps }: OverridingProps) => (
        <ExportItemList {...overridingProps} />
    );

    it("should display all todo items", () => {
        const todoItems = [createTodoItem("Milk"), createTodoItem("Bread")];

        const wrapper = shallow(exportItemList({ todoItems }));

        expect(wrapper).toMatchSnapshot();
    });
});
