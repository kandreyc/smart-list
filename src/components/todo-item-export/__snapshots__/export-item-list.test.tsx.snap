// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportItemList should display all todo items 1`] = `
<Fragment>
  <ExportItemView
    exportItem={
      ExportItem {
        "category": null,
        "generalName": "Milk",
        "id": 1,
        "isBought": true,
        "productPrice": null,
        "purchasedPrice": null,
        "quantity": 1,
        "targetProduct": null,
      }
    }
    key="1"
  />
  <ExportItemView
    exportItem={
      ExportItem {
        "category": null,
        "generalName": "Bread",
        "id": 2,
        "isBought": true,
        "productPrice": null,
        "purchasedPrice": null,
        "quantity": 1,
        "targetProduct": null,
      }
    }
    key="2"
  />
</Fragment>
`;
